### ISSUE TITLE
[CI] CompositeAggregatorTests.testEarlyTermination

### ISSUE THREAD

**Issue by andreidan** [elastic/elasticsearch#62769]
**Build scan**: 
https://gradle-enterprise.elastic.co/s/hmrpc46ciypww

**Repro line**:
```
./gradlew ':server:test' --tests "org.elasticsearch.search.aggregations.bucket.composite.CompositeAggregatorTests.testEarlyTermination" \
  -Dtests.seed=74C000FFBDD6744 \
  -Dtests.security.manager=true \
  -Dtests.locale=sr-Latn-ME \
  -Dtests.timezone=Africa/Kigali \
  -Druntime.java=8
```

**Reproduces locally?**: 
Yes

**Applicable branches**: 
`7.x`

**Failure history**:

**Failure excerpt**:
```
java.lang.AssertionError
	at __randomizedtesting.SeedInfo.seed([74C000FFBDD6744:C8DB4150C3B1F747]:0)
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at org.elasticsearch.search.aggregations.bucket.composite.CompositeAggregatorTests.lambda$testEarlyTermination$158(CompositeAggregatorTests.java:2151)
	at org.elasticsearch.search.aggregations.bucket.composite.CompositeAggregatorTests.executeTestCase(CompositeAggregatorTests.java:2281)
	at org.elasticsearch.search.aggregations.bucket.composite.CompositeAggregatorTests.testEarlyTermination(CompositeAggregatorTests.java:2136)
```

**Comment by cbuescher** [comment]
Recent failure on 7.10: https://gradle-enterprise.elastic.co/s/2jvlobukahkb2

**Comment by davidkyle** [comment]
Another, this time on 7.9. https://gradle-enterprise.elastic.co/s/xuonekc5ocmgi

```
java.lang.AssertionError
	at __randomizedtesting.SeedInfo.seed([5D595BA386DA7836:92CE1AFCBEB6E835]:0)
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at org.elasticsearch.search.aggregations.bucket.composite.CompositeAggregatorTests.lambda$testEarlyTermination$134(CompositeAggregatorTests.java:1920)
	at org.elasticsearch.search.aggregations.bucket.composite.CompositeAggregatorTests.executeTestCase(CompositeAggregatorTests.java:2069)
	at org.elasticsearch.search.aggregations.bucket.composite.CompositeAggregatorTests.testEarlyTermination(CompositeAggregatorTests.java:1905)
```

**Comment by nik9000** [comment]
I checked out the commit hash gradle mentions and couldn't reproduce locally. If anyone can reproduce locally can they ping me? I haven't dug *too* deep but it looks like something that I'm going to need to consistently reproduce to debug.
```
git checkout 0b6caff45fdc33cc4a5c9ce37b9161c0ace289a9
 ./gradlew ':server:test' --tests "org.elasticsearch.search.aggregations.bucket.composite.CompositeAggregatorTests.testEarlyTermination"  \
  -Dtests.seed=5D595BA386DA7836 -Dtests.security.manager=true -Dtests.locale=es-ES \
  -Dtests.timezone=Israel -Druntime.java=8 -Dtests.fips.enabled=true
```

**Comment by pugnascotia** [comment]
Failed again: https://gradle-enterprise.elastic.co/s/h4gfwgqmxsdiw

@nik9000 I was able to reproduce locally with:

```
./gradlew ':server:test' --tests "org.elasticsearch.search.aggregations.bucket.composite.CompositeAggregatorTests.testEarlyTermination" -Dtests.seed=8230A57727FF591C -Dtests.security.manager=true -Dtests.locale=tr -Dtests.timezone=Pacific/Auckland -Druntime.java=8
```

**Comment by nik9000** [comment]
> @nik9000 I was able to reproduce locally with:

I can too! I'll dig.

**Comment by martijnvg** [comment]
I'm backporting Nik's fix to the 7.10 branch as this CI failure is still occurring there: https://gradle-enterprise.elastic.co/s/qq4q5ao35y6oe

**Comment by nik9000** [comment]
Oh no, sorry! I hadn't realized I missed a branch.
